src/charge/modifiers/exception/OnceException.php

```diff
@@ -0,0 +1,9 @@
+<?php declare(strict_types=1);
+
+namespace hiqdev\php\billing\charge\modifiers\exception;
+
+use Exception;
+
+class OnceException extends Exception
```

Comment:
In general, it makes sense to distinguish compile-time and runtime exceptions.

A compile-time exception happens during creation and does not require a charge: wrong syntax, passed fraction of month, etc. I'd not create a separate exception class for compile-time errors and instead use `FormulaEngineException` or `FormulaSemanticsError`.

Runtime exceptions occur when you apply a valid formula to a charge and meet some constraints. Use either [InvariantException](https://github.com/hiqdev/php-billing/-/blob/master/src/Exception/InvariantException.php) or [ConstraintException](https://github.com/hiqdev/php-billing/-/blob/master/src/Exception/ConstraintException.php)